### PR TITLE
General repository cleanup

### PR DESCRIPTION
**Pull request by yaziciahmet**
# Description

General cleanup of the repository. Removes unused crates & dependencies, but doesn't change the way anything works. Even if something is not used feature-wise, but is well integrated to the existing code, it is not touched.

Also ran cargo update to clear up Cargo.lock.

Drops build crate number to 900, and release crate number to 1005.

## Linked Issues
- Fixes # (issue, if applicable)
- Related to # (issue) 

## Testing
Describe how these changes were tested. If you've added new features, have you added unit tests?

## Docs
Describe where this code is documented. If it changes a documented interface, have the docs been updated?

### CODE REVIEW

File: crates/sovereign-sdk/rollup-interface/specs/overview.md
Comment (eyusufatik) [inline crates/sovereign-sdk/rollup-interface/specs/overview.md:1]: i think deleted files under rollup-interface/specs should stay.

they might come in handy for people starting to work on this repo.

**Review by eyusufatik — CHANGES_REQUESTED** [review]
other than my single comment pr lgtm

in the past I wanted to keep as many crates as possible since the tests inside them might help us find some bugs we put in the repo in the future, but it's becoming too much of a burden.

File: crates/sovereign-sdk/rollup-interface/specs/overview.md
Comment (yaziciahmet) [reply crates/sovereign-sdk/rollup-interface/specs/overview.md:1]: done

File: bin/citrea/provers/risc0/batch-prover-mock/Cargo.lock [hunk 1]
```diff
@@ -721,9 +721,9 @@ dependencies = [
 
 [[package]]
 name = "brotli"
-version = "6.0.0"
+version = "7.0.0"
```
Comment (eyusufatik) [inline bin/citrea/provers/risc0/batch-prover-mock/Cargo.lock]: is brotliv7 compatible with brotliv6?

File: crates/sovereign-sdk/module-system/module-implementations/sov-bank/Cargo.toml [hunk 2]
```diff
@@ -50,3 +50,6 @@ native = [
 cli = ["native"]
 serde = []
 test-utils = []
+
+[package.metadata.cargo-udeps.ignore]
```
Comment (eyusufatik) [inline crates/sovereign-sdk/module-system/module-implementations/sov-bank/Cargo.toml:54]: why ignore instead of removing?

Comment (yaziciahmet) [reply crates/sovereign-sdk/module-system/module-implementations/sov-bank/Cargo.toml:54]: explained in private

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**